presto-main/src/main/java/com/facebook/presto/operator/annotations/ImplementationDependency.java

```diff
@@ -15,9 +15,64 @@
 
 import com.facebook.presto.metadata.BoundVariables;
 import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.spi.function.FunctionDependency;
+import com.facebook.presto.spi.function.OperatorDependency;
+import com.facebook.presto.spi.function.TypeParameter;
 import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.type.LiteralParameter;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Set;
+
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
 
 public interface ImplementationDependency
 {
     Object resolve(BoundVariables boundVariables, TypeManager typeManager, FunctionRegistry functionRegistry);
+
+    static boolean isImplementationDependencyAnnotation(Annotation annotation)
+    {
+        return annotation instanceof TypeParameter ||
+                annotation instanceof LiteralParameter ||
+                annotation instanceof FunctionDependency ||
+                annotation instanceof OperatorDependency;
+    }
+
+    public static class Factory
+    {
+        private Factory()
+        {}
+
+        public static ImplementationDependency createDependency(Annotation annotation, Set<String> literalParameters)
```

Comment:
I don't feel like it's necessary to add have `Factory` class. This method can be directly in the interface. I'll revisit later.